layers/+chat/circe/packages.el

```diff
@@ -0,0 +1,142 @@
+;;; packages.el --- Circe layer packages file for Spacemacs.
+;;
+;; Copyright (c) 2012-2017 Sylvain Benner & Contributors
+;;
+;; Author: Chris Barrett <chris.d.barrett@me.com>
+;; Author: Aidan Nyquist <contact@aidannyquist.com>
+;; URL: https://github.com/syl20bnr/spacemacs
+;;
+;; This file is not part of GNU Emacs.
+;;
+;;; License: GPLv3
+
+;;; Code:
+
+(defconst circe-packages
+  '(circe
+    circe-notifications
+    persp-mode
+    helm
+    helm-circe
+    ))
+
+(defun circe/init-circe ()
+  (use-package circe
+    :commands circe
+    :config
+
+    (set-face-background 'circe-prompt-face nil)
+    (setq circe-reduce-lurker-spam t)
+    (setq circe-use-cycle-completion t)
+    (setq circe-active-users-timeout (* 60 30)) ; 30 minutes
+    (setq circe-format-say "{nick}> {body}")
+    (setq circe-format-self-say (concat (propertize ">>>" 'face 'circe-self-say-face) " {body}"))
+    (setq circe-prompt-string (concat (propertize ">>>" 'face 'circe-prompt-face) " "))
+    (setq circe-highlight-nick-type 'all)
+
+    ;; Add circe-color-nicks
+    (enable-circe-color-nicks)
+    (setq circe-color-nicks-everywhere 't)
+
+    ;; Show channel name in prompt.
+    (add-hook 'circe-chat-mode-hook
+              (lambda ()
+                (-let* (
+                        ;; Get buffername
+                        (s (buffer-name))
+                        ((_ gitter-fmt) (s-match (rx (+ nonl) "/" (group (+ nonl)) eos) s))
+                        ((_ standard-fmt) (s-match (rx "#" (group (+ nonl)) eos) s))
+                        ;; Buffername to channel name
+                        (chan (or gitter-fmt standard-fmt))
+                        ;; Format prompt
+                        (prompt (concat "#" (propertize chan 'face 'circe-prompt-face) " > ")))
+                  (lui-set-prompt prompt))))
+
+    ;; Timestamps in margins.
+    (setq lui-time-stamp-position 'right-margin)
+    (setq lui-time-stamp-format "%H:%M")
+    (setq lui-fill-type nil)
+    (add-hook 'lui-mode-hook (lambda ()
```

Comment:
Better to add a named function to the layer's `funcs.el` and use that instead of a lambda.